src/averageAges.js

```diff
@@ -55,10 +69,32 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter(
+    child => {
+      const mother = people.find(person => child['mother'] === person['name']);
+
+      return onlyWithSon
+        ? mother && child['sex'] === 'm'
```

Comment:
* Using dynamic keys is redundant, consider using static keys instead.
* Avoid using magic strings - it's a programming antipattern. Consider using constants with meaningful names instead.
* Follow the DRY (don't repeat yourself) principle. Consider using helper functions for repeating functionality